pandas/tests/io/parser/test_dtypes.py

```diff
@@ -509,3 +509,27 @@ def test_numeric_dtype(all_parsers, dtype):
 
     result = parser.read_csv(StringIO(data), header=None, dtype=dtype)
     tm.assert_frame_equal(expected, result)
+
+
+def test_intna_precision(all_parsers):
+    parser = all_parsers
+    data = "1556559573141592653\n1556559573141592654\n\n1556559573141592655"
+    dtype = 'Int64'
+
+    expected = DataFrame([
+        [1556559573141592653],
+        [1556559573141592654],
+        [0],
```

Comment:
Just to clarify the below TODO if you use `np.nan` the precision is lost before the DataFrame gets constructed, hence why I've constructed this with just integers first and subsequently assigned nan here